bc_obps/registration/models/operator.py

```diff
@@ -0,0 +1,123 @@
+import uuid
+from django.db import models
+from registration.constants import (
+    BC_CORPORATE_REGISTRY_REGEX,
+    BC_CORPORATE_REGISTRY_REGEX_MESSAGE,
+    CRA_BUSINESS_NUMBER_MESSAGE,
+)
+from registration.models import TimeStampedModel, Address, BusinessStructure, Contact, Document, User
+from simple_history.models import HistoricalRecords
+from django.core.validators import RegexValidator, MaxValueValidator, MinValueValidator
+
+
+class Operator(TimeStampedModel):
+    """Operator model"""
+
+    class Statuses(models.TextChoices):
+        DRAFT = "Draft"
+        PENDING = "Pending"
+        APPROVED = "Approved"
+        DECLINED = "Declined"
+        CHANGES_REQUESTED = "Changes Requested"
+
+    id = models.UUIDField(
+        primary_key=True, default=uuid.uuid4, db_comment="Primary key to identify the operator", verbose_name="ID"
+    )
+    legal_name = models.CharField(max_length=1000, db_comment="The legal name of an operator", unique=True)
+    trade_name = models.CharField(max_length=1000, blank=True, db_comment="The trade name of an operator")
+    cra_business_number = models.IntegerField(
+        validators=[
+            MaxValueValidator(999999999, message=CRA_BUSINESS_NUMBER_MESSAGE),
+            MinValueValidator(100000000, message=CRA_BUSINESS_NUMBER_MESSAGE),  # Assuming  9-digit number
+        ],
+        db_comment="The CRA business number of an operator",
+    )
+    swrs_organisation_id = models.IntegerField(
+        db_comment="An identifier used in the CIIP/SWRS dataset (in swrs: organisation = operator). This identifier will only be populated for operators that were imported from that dataset.",
+        blank=True,
+        null=True,
+    )
+    bc_corporate_registry_number = models.CharField(
+        db_comment="The BC corporate registry number of an operator",
+        null=True,
+        unique=True,
+        validators=[RegexValidator(regex=BC_CORPORATE_REGISTRY_REGEX, message=BC_CORPORATE_REGISTRY_REGEX_MESSAGE)],
+    )
+    business_structure = models.ForeignKey(
+        BusinessStructure,
+        on_delete=models.DO_NOTHING,
+        null=True,
+        blank=True,
+        db_comment="The business structure of an operator",
+        related_name="operators",
+    )
+    physical_address = models.ForeignKey(
+        Address,
+        on_delete=models.DO_NOTHING,
+        null=True,
+        db_comment="The physical address of an operator (where the operator is physically located)",
+        related_name="operators_physical",
+    )
+    mailing_address = models.ForeignKey(
+        Address,
+        on_delete=models.DO_NOTHING,
+        db_comment="The mailing address of an operator",
+        related_name="operators_mailing",
+        blank=True,
+        null=True,
+    )
+    website = models.URLField(
```

Comment:
It could be a separate PR, because we need to modify endpoints, schemas, tests and ...
I can work on it if you give me a list of all the fields we need to remove.

Reply:
Let's make it a separate ticket (I'll create it). I suspect there's no need to rush into removing these fields.

Reply:
created ticket #1765 to capture this work

Reply:
Thank you! I'm going to add the below task to your ticket:
- Make the `Operation` data model type field an ENUM and populate it using production data.